core/src/main/java/com/linecorp/armeria/common/metric/MeterIdFunction.java

```diff
@@ -0,0 +1,124 @@
+/*
+ *  Copyright 2017 LINE Corporation
+ *
+ *  LINE Corporation licenses this file to you under the Apache License,
+ *  version 2.0 (the "License"); you may not use this file except in compliance
+ *  with the License. You may obtain a copy of the License at:
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ *  WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ *  License for the specific language governing permissions and limitations
+ *  under the License.
+ */
+
+package com.linecorp.armeria.common.metric;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.List;
+import java.util.function.BiFunction;
+
+import com.google.common.base.MoreObjects;
+import com.google.common.collect.ImmutableList;
+
+import com.linecorp.armeria.client.metric.MetricCollectingClient;
+import com.linecorp.armeria.common.HttpHeaders;
+import com.linecorp.armeria.common.RequestContext;
+import com.linecorp.armeria.common.RpcRequest;
+import com.linecorp.armeria.common.logging.RequestLog;
+import com.linecorp.armeria.server.PathMapping;
+import com.linecorp.armeria.server.ServiceRequestContext;
+import com.linecorp.armeria.server.metric.MetricCollectingService;
+
+import io.micrometer.core.instrument.MeterRegistry;
+import io.micrometer.core.instrument.Tag;
+import io.micrometer.core.instrument.Tags;
+import io.micrometer.core.instrument.util.MeterId;
+
+/**
+ * Creates a {@link MeterId} from a {@link RequestLog}.
+ *
+ * @see MetricCollectingClient
+ * @see MetricCollectingService
+ */
+@FunctionalInterface
+public interface MeterIdFunction {
+
+    /**
+     * Returns the default function that creates a {@link MeterId} with the specified name and the {@link Tag}s
+     * derived from the current {@link PathMapping} (if available) and HTTP (or RPC) method name.
+     * e.g. {@code my_service_name{pathMapping="exact:/service/path",method="POST"}}
+     */
+    static MeterIdFunction ofDefault(String firstNamePart, String... otherNameParts) {
+        requireNonNull(firstNamePart, "firstNamePart");
+        requireNonNull(otherNameParts, "otherNameParts");
+
+        final List<String> name =
+                ImmutableList.<String>builder().add(firstNamePart).add(otherNameParts).build();
+
+        return (registry, log) -> {
+            final RequestContext ctx = log.context();
+            final Object requestEnvelope = log.requestHeaders();
```

Comment:
Just a cruft from the old days when `log.requestHeaders()` was `log.requestEnvelope()`. Fixed.